### PR TITLE
Add configuration for maven repository

### PR DESCRIPTION
**Pull request by jasonzhu1991**

### CODE REVIEW

**Comment by jasonzhu1991** [comment]
@unkdrew @chanyeeham @chenjh68 Please review.

File: pom.xml [hunk 1]
```diff
@@ -16,4 +22,54 @@
...
+    <repository>
+      <id>internal.repo</id>
+      <name>Temporary Staging Repository</name>
+      <url>file://${project.build.directory}/mvn-repo</url>
+    </repository>
+  </distributionManagement>
+
+  <build>
+    <pluginManagement>
+      <plugins>
```
Comment (unkdrew) [inline pom.xml]: This <pluginManagement/plugins> is different from the <plugins> below, right?

File: pom.xml [hunk 2]
```diff
@@ -4,6 +4,12 @@
   <artifactId>priceninja-servicemodel</artifactId>
   <version>0.0.1-SNAPSHOT</version>
   <name>PriceNinjaServiceModel</name>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <github.global.server>github</github.global.server>
```
Comment (jasonzhu1991) [inline pom.xml]: Dummy Comment

File: pom.xml [hunk 3]
```diff
@@ -16,4 +22,54 @@
...
+      </plugins>
+    </pluginManagement>
+    <!-- To use the plugin goals in your POM or parent POM -->
+    <plugins>
+      <plugin>
+        <groupId>com.github.github</groupId>
+        <artifactId>site-maven-plugin</artifactId>
+        <version>0.11</version>
+        <configuration>
+          <message>Maven artifacts for ${project.version}</message>  <!-- git commit message -->
```
Comment (unkdrew) [inline pom.xml]: Can the spacing before comments be consistent? I would rather start the comment on the line above the code in this case.

**Comment by jasonzhu1991** [comment]
@unkdrew Made a few updates. Maven deploy plugin in PluginManagement wasn't actually needed so I removed it.

**Comment by jasonzhu1991** [comment]
@unkdrew Squashed commits. Excluded .DS_Store when deploying to maven repository, and use merging instead of replacing when deploying to maven repo.

**Pull request merged**